src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`2.cs

```diff
@@ -121,6 +121,23 @@ public void AddWork(IEnumerable<TItem> items)
 
             return;
 
+            void AddItemsToBatch(IEnumerable<TItem> items)
+            {
+                // no equality comparer.  We want to process all items.
+                if (_equalityComparer == null)
+                {
+                    _nextBatch.AddRange(items);
+                    return;
+                }
+
+                // We're deduping items.  Only add the item if it's the first time we've seen it.
+                foreach (var item in items)
+                {
+                    if (_uniqueItems.Add(item))
+                        _nextBatch.Add(item);
+                }
+            }
```

Comment:
this was just a move for clarity.  This needs to be called under a lock.  so i moved this into the only method that calls this to make sure no one else accidentally calls this incorrectly outsdie of hte lock.